src/entities/Snapshot/utils.ts

```diff
@@ -87,3 +91,65 @@ export function groupProposalsByMonth(proposals: Partial<SnapshotProposal>[], fi
 
   return data
 }
+
+export async function getSnapshotStatusAndSpace(spaceName?: string) {
+  spaceName = spaceName && spaceName.length > 0 ? spaceName : SNAPSHOT_SPACE
+  const values = await Promise.all([
+    await SnapshotGraphql.get().getStatus(),
+    await SnapshotGraphql.get().getSpace(spaceName),
+  ])
+  const snapshotStatus = values[0]
+  const snapshotSpace = values[1]
+  if (!snapshotSpace) {
+    throw new Error(`Couldn't find snapshot space ${spaceName}. 
+      \nSnapshot response: ${JSON.stringify(snapshotSpace)}
+      \nSnapshot status: ${JSON.stringify(snapshotStatus)}`)
+  }
+  return { snapshotStatus, snapshotSpace }
+}
+
+export function filterDelegationTo(delegations: Delegation[], space: string): Delegation[] {
+  if (delegations.length > 1) {
+    return delegations.filter((delegation) => delegation.space === space)
+  }
+
+  return delegations
+}
+
+export function filterDelegationFrom(delegations: Delegation[], space: string): Delegation[] {
+  if (delegations.length === 0) {
+    return []
+  }
+
+  const unique_delegations = new Map<string, Delegation>()
+
+  for (const deleg of delegations) {
+    if (unique_delegations.has(deleg.delegator)) {
+      if (unique_delegations.get(deleg.delegator)?.space !== space) {
+        unique_delegations.set(deleg.delegator, deleg)
+      }
+    } else {
+      unique_delegations.set(deleg.delegator, deleg)
+    }
+  }
+
+  return Array.from(unique_delegations.values())
+}
+
+export async function fetchAndFilterDelegates(query: string, variables: any): Promise<DelegationResult> {
```

Comment:
Shouldn't this be part of `SnapshotService` or is it because it is used by the frontend?

Reply:
Also I'd rename this to `getFilteredDelegates` (but it is not part of this PR)

Reply:
Cannot add it to a snapshot service, but refactored it a little bit more :)